### ISSUE TITLE
Validaciones en la carga

### ISSUE THREAD

**Issue by dariorodriguez** [INTA-Suelos/SiSinta#82]
Voy a ir armando acá un listado de valores posibles para los parámetros de suelo, de modo que no aparezcan datos incoherentes como pHs de 73

**Comment by midraed** [comment]
Creo que lo que se peude generar es para cada campo analitico (porque estamos hablando solo de los campos analiticos) un csv con nombre_del_campo, valor_minimo_posible, valor_maximo_posible, valor_minimo_esperable, valor_maximo_esperable.

Por ejemplo (me invento los esperables)
analitico_ph_pasta, 0, 14, 3, 12 

Y eso quizas podria servir para marcar datos erroneos, y datos dudosos. Incluso podria estar integrado en la visualizacion de la ficha:

![ph](https://cloud.githubusercontent.com/assets/2235494/6147404/ed1746ec-b1d5-11e4-8a5e-81f70f1a6636.png)

Si un archivo o asi te sirve @mauriciopasquier, yo puedo encargarme de armar en mi fork de desarrollo, o directamente en este, el archivo base con todos los campos, y explicarle a @dariorodriguez como editarlo e irle agregando lo valores.

Claro que a veces van a existir los valores minimos y maximos posibles y en algunos casos raros, no. Y los valores esperables.. es mas dificil, porque no siempre existen. O los suelos de uno u otro lugar pueden ser muy diferentes.

**Comment by dariorodriguez** [comment]
Maravilloso. Totalmente de acuerdo

El 11 de febrero de 2015, 10:11, Guillermo Federico Olmedo <
notifications@github.com> escribió:

> Creo que lo que se peude generar es para cada campo analitico (porque
> estamos hablando solo de los campos analiticos) un csv con
> nombre_del_campo, valor_minimo_posible, valor_maximo_posible,
> valor_minimo_esperable, valor_maximo_esperable.
>
> Por ejemplo (me invento los esperables)
> analitico_ph_pasta, 0, 14, 3, 12
>
> Y eso quizas podria servir para marcar datos erroneos, y datos dudosos.
> Incluso podria estar integrado en la visualizacion de la ficha:
>
> [image: ph]
> <https://cloud.githubusercontent.com/assets/2235494/6147404/ed1746ec-b1d5-11e4-8a5e-81f70f1a6636.png>
>
> Si un archivo o asi te sirve @mauriciopasquier
> <https://github.com/mauriciopasquier>, yo puedo encargarme de armar en mi
> fork de desarrollo, o directamente en este, el archivo base con todos los
> campos, y explicarle a @dariorodriguez <https://github.com/dariorodriguez>
> como editarlo e irle agregando lo valores.
>
> Claro que a veces van a existir los valores minimos y maximos posibles y
> en algunos casos raros, no. Y los valores esperables.. es mas dificil,
> porque no siempre existen. O los suelos de uno u otro lugar pueden ser muy
> diferentes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/INTA-Suelos/SiSinta/issues/82#issuecomment-73878295>.
>



-- 
Darío

**Comment by mauriciopasquier** [comment]
@midraed si querés ir armandolo así, buenísimo. Pero dudo que el csv sea la forma final en la que se use si son valores que no van a cambiar (los mínimos y máximos al menos). Sobre los esperados, ustedes sabrán si vale la pena. Se puede usar lo que haya y si no hay ignorar ese chequeo.

**Comment by midraed** [comment]
@mauriciopasquier no, creo que no son valores que vayan a cambiar, una vez acordemos. Si puede pasar que en el futuro se rediscuta algo, y se mueva un poco un valor esperable.. pero no deberia ser algo comun y que por lo tnato necesite facil actualizacion.
Lo del csv, es para que te sea mas facil tomarlo, vs. tomarlo del texto de la discusion del issue. Pero pensandolo bien te serviria mas en una semilla, como el formato de las ayudas? Tipo:

    - id: 1
    campo: analitico_ph_pasta
    min_pos: 0
    max_pos: 14
    min_esp: 3
    max_esp: 12

o es directamente algo que meterias en el codigo, y de cualquier manera te viene bien porque va a terminar en copy paste?
Avisame que te viene mejor. 

Con respecto a los valores esperables, si, deberian ser opcionales.

Y ahora pregunta de git/github:
Lo mas sencillo seria hacer ese archivo en INTA-Suelos/SiSinta: desarrollo, pero si queres lo hago en midraed/SiSinta: desarrollo y te hago un pull request. Pero si el que va a editar es dario.. con solo ponerlo de colaborador en mi repo alcanza? 
Tambien podemos armar otr branch en INTA-Suelos... pero me cuesta decidir que es lo mas correcto.
Dario podria hacer un fork.. pero lo vamos a complicar al cuete.

**Comment by mauriciopasquier** [comment]
Para mí es algo que va a terminar en el código (por las validaciones, por ejemplo) de una u otra forma. Asique no me preocuparía por el formato en el que lo arman, lo veo más como documentación por ahora. Por la misma razón, no creo que sea necesario meterlo como archivo en el repositorio. Tal vez sea mejor hacer una tabla acá en el wiki: https://github.com/INTA-Suelos/SiSinta/wiki/Valores-posibles

**Comment by midraed** [comment]
Ok.. excelente. Ahí agregué toda la lista de campos analiticos. @dariorodriguez te animas a 
completarla? 
Tenes que apretar edit, y luego poner los valores entre las barritas. Cualquier cosa: me decis!

**Comment by dariorodriguez** [comment]
Esto te va a costar un par de birras

El 11 de febrero de 2015, 11:27, Guillermo Federico Olmedo <
notifications@github.com> escribió:

> Ok.. excelente. Ahí agregué toda la lista de campos analiticos.
> @dariorodriguez <https://github.com/dariorodriguez> te animas a
> completarla?
> Tenes que apretar edit, y luego poner los valores entre las barritas.
> Cualquier cosa: me decis!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/INTA-Suelos/SiSinta/issues/82#issuecomment-73888936>.
>



-- 
Darío

**Comment by midraed** [comment]
no hay problema.

**Comment by dariorodriguez** [comment]
Bueno...eso ya lo completé
Consulté a Daniel Carreira para datos de laboratorio, a Patricia Carfagno para valores de constantes hídricas y Romina Romaniuk me dió una mano con algunos valores que tenía dudas (consultó a la gente del laboratorio de suelos de Edafo-UBA)
Creo que ya está listo
